File: mobile/ios/EmbeddedCounselDemo/EmbeddedCounselDemo/ResponseModels.swift
//
//  ResponseModels.swift
//  EmbeddedCounselDemo
//
//  Created by Counsel on 5/3/25.
//

import Foundation

struct SignedAppResponse: Codable {
    let url: String
}

enum UserType: String, Codable {
    case main
    case onboarding
}

struct AccessCodeTokenResponse: Codable {
    let token: String
    let userType: UserType
}

File: mobile/android/EmbeddedCounselDemo/app/src/main/java/health/counsel/embeddeddemo/data/api/ResponseModels.kt
package health.counsel.embeddeddemo.data.api

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
enum class UserType {
    @SerialName("main") MAIN,
    @SerialName("onboarding") ONBOARDING,
}

@Serializable
data class AccessCodeTokenResponse(
    val token: String,
    val userType: UserType,
)

@Serializable
data class SignedAppResponse(
    val url: String,
)
